patches/kaggle_module_resolver.py

```diff
@@ -7,7 +7,7 @@
 url_pattern = re.compile(r"https?://([a-z]+\.)?kaggle.com/models/(?P<owner>[^\\/]+)/(?P<model>[^\\/]+)/frameworks/(?P<framework>[^\\/]+)/variations/(?P<variation>[^\\/]+)/versions/(?P<version>[0-9]+)$")
 
 def _is_on_kaggle_notebook():
-    return os.getenv("KAGGLE_KERNEL_RUN_TYPE") != None
+    return os.getenv("KAGGLE_KERNEL_RUN_TYPE") != None and os.getenv("KAGGLE_USER_SECRETS_TOKEN") != None
```

Comment:
Great, I also looked that the unit test is already setting this value: https://github.com/Kaggle/docker-python/blob/a1503982e1fd69227222b950e888dafbdb45fc78/tests/test_kaggle_module_resolver.py#L26

Which explains why the tests are passing.